src/middlewares/admin.js

```diff
@@ -0,0 +1,14 @@
+
+const isAdmin = function (req, res, next) {
+    if (req.session.user) {
+        if (req.session.user.role === 'ADMIN') {
```

Comment:
Ensure that the role check is comprehensive and accounts for cases where req.session.user might be undefined or manipulated.